compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -356,6 +356,7 @@ impl<'mir, 'tcx> Checker<'mir, 'tcx> {
             };
 
             match *ty.kind() {
+                ty::FnPtr(_) => break,
```

Comment:
This will cause `(fn(), &mut u32)` types to not error (please add a test for it). You will need to rewrite the entire `for` loop and use a `TypeVisitor` instead. If you'd like more instructions than this hint, let me know, I'll write something up

Reply:
Thanks, I completely missed that. 

Added test and updated to use `TypeVisitor`. My implementation follows how other implementations I found in `rustc` work.
Please let me know if anything else needs to change.

Reply:
Looks great to me now!